### PR TITLE
[Statepoint] Return undef value for the statepoint of the none token

### PR DESCRIPTION
**Pull request by danilaml**
Helps avoid the crash in verifier when it tries to print the error. `none` token might be produced by llvm-reduce, since it's a default value, so by extension this also fixes llvm-reduce crash, allowing it to just discard invalid IR.

We might special-case tokens in ReduceArguments and to use some other value, other than `getDefaultValue` (or change the `getDefaultValue` to use something other than `getNullValue` for token), but it still wouldn't fix the fact that verifier would crash trying to print an error message if it ever encounters such relocation.
I wasn't able to come up with any potential issues with handling `none` token as if it was `undef`, but it doesn't mean that there isn't any =)

### CODE REVIEW

File: llvm/lib/IR/IntrinsicInst.cpp [hunk 1]
```diff
@@ -870,6 +870,11 @@ const Value *GCProjectionInst::getStatepoint() const {
   if (isa<UndefValue>(Token))
     return Token;
 
+  // Treat none token as if it was undef here
+  if (isa<ConstantTokenNone>(Token)) {
+    return UndefValue::get(Token->getType());
+  }
```
Comment (arpilipe) [inline llvm/lib/IR/IntrinsicInst.cpp:876]: Nit.
```suggestion
  if (isa<ConstantTokenNone>(Token))
    return UndefValue::get(Token->getType());
```

**Review by arpilipe — APPROVED** [review]

**Comment by nunoplopes** [comment]
Can you use poison here?
We are trying to get rid of undef, so if this can use poison today, it would help the effort.
Thank you!

**Comment by danilaml** [comment]
@nunoplopes possibly. Not sure what semantics of this would be though. And this would need to be adapted to poison as well (otherwise there is no point): https://github.com/danilaml/llvm-project/commit/d693fd29f1885e72b90428999aee92573abb9c2d

**Comment by nunoplopes** [comment]
> @nunoplopes possibly. Not sure what semantics of this would be though. And this would need to be adapted to poison as well (otherwise there is no point): [d693fd2](https://github.com/llvm/llvm-project/commit/d693fd29f1885e72b90428999aee92573abb9c2d)

Right. It seems all of those could be converted to poison. It's being used just as a placeholder, so anything works.

**Pull request merged**